lib/core/yat_service.dart

```diff
@@ -35,9 +37,12 @@ class YatService {
       final resBody = json.decode(response.body) as Map<String, dynamic>;
 
       final results = resBody["result"] as Map<dynamic, dynamic>;
-      results.forEach((dynamic key, dynamic value) {
-        yatRecords.add(YatRecord.fromJson(value as Map<String, dynamic>));
-      });
+   
+      // Favour a subaddress over a standard address.
+      final yatRecord = results[MONERO_SUB_ADDRESS] ?? results[MONERO_STD_ADDRESS];
+      if (yatRecord != null) {
+        yatRecords.add(YatRecord.fromJson(value as Map<String, dynamic>))
+      }
```

Comment:
Here is unclean from where come `value`.

Reply:
Maybe you meant `yatRecord` instead of `value` ?

Reply:
And as i see that `fetchYatAddress` will works only for `xmr` or i miss something ?

Reply:
> And as i see that `fetchYatAddress` will works only for `xmr` or i miss something ?

It will fetch whatever tags are passed to it, based on if the user is looking up XMR / BTC / LTC. If nothing is passed in it will return all attached wallet addresses.

However if the user has added a standard and a sub address, then you will get back 2 results.